taichi/aot/module_loader.h

```diff
@@ -37,34 +38,35 @@ class TI_DLL_EXPORT Kernel {
   virtual void launch(RuntimeContext *ctx) = 0;
 };
 
-class TI_DLL_EXPORT ModuleLoader {
+class TI_DLL_EXPORT Module {
  public:
   // Rule of 5 to make MSVC happy
-  ModuleLoader() = default;
-  virtual ~ModuleLoader() = default;
-  ModuleLoader(const ModuleLoader &) = delete;
-  ModuleLoader &operator=(const ModuleLoader &) = delete;
-  ModuleLoader(ModuleLoader &&) = default;
-  ModuleLoader &operator=(ModuleLoader &&) = default;
-
-  // TODO: Add method get_kernel(...) once the kernel field data will be
-  // generic/common across all backends.
-
-  virtual bool get_field(const std::string &name,
-                         aot::CompiledFieldData &field) = 0;
-
-  /**
-   * @brief Get the kernel object
-   *
-   * @param name Name of the kernel
-   * @return Kernel*
-   */
-  Kernel *get_kernel(const std::string &name);
-
+  Module() = default;
+  virtual ~Module() = default;
+  Module(const Module &) = delete;
+  Module &operator=(const Module &) = delete;
+  Module(Module &&) = default;
+  Module &operator=(Module &&) = default;
+
+  static std::unique_ptr<Module> load(const std::string &path,
+                                      Arch arch,
+                                      std::any mod_params);
+  // TODO
+  // Module metadata
+  // Arch arch() const;
+  // uint64_t version() const;
+
+  // TODO
+  // APIs to be overriden by each backend.
+  // virtual std::unique_ptr<Field> get_field(const std::string &name) = 0;
+  // virtual std::unique_ptr<Kernel> get_kernel(const std::string &name) = 0;
```

Comment:
Why not implement this?

Reply:
Try to make this PR small to only contain one change, as the title says. This will be implemented to replace the old one in derived classes.

Reply:
`get_kernel` already exists in the current codebase, let's just add it back.

Reply:
signatures are different. but ok, will do